Dockerfile & Dockercompose file

aqeelmozumder opened the pull request:
The Dockercompose file pulls postgres image and awaits for connection.

The dockerfile pulls image from node and copies the package.json and run dependencies and exposes port 4000

aomi commented on dockerfile:
```diff
@@ -0,0 +1,11 @@
+FROM node
```
Usually called `Dockerfile` instead of `dockerfile`.
Also I would pick a version of Node here (probably 16) `node:16-alpine` and Alpine for smaller image sizes.

aomi commented on dockerfile:
```diff
@@ -0,0 +1,11 @@
+FROM node
+
+WORKDIR /usr/src/app
+COPY package.json ./
```
You need the lockfile here as well.

aomi reviewed (commented):
Some general points about Docker containers for Node projects:
https://snyk.io/blog/10-best-practices-to-containerize-nodejs-web-applications-with-docker/

TypeScript specific Dockerfile tutorial:
https://towardsdev.com/writing-a-docker-file-for-your-node-js-typescript-micro-service-c5170b957893?gi=1e08f62406e4

MustafaWasif commented:
I ran into a build issue while trying to create an image. I modified the dockerfile and now seems to work. 
Basically added a `npm run build `before `npm start`.
[https://github.com/SENG499-S22-Company3/backend/blob/mustafa/dockerfile-modified/dockerfile](url)

The build script in package.json was also modified to be able to copy schema.graphql file to destination folder after compilation (i.e., .ts to .js).
`"build":` `"tsc -p tsconfig.json && copyfiles --flat src/schema/*.graphql [build/src/schema/"`
[https://github.com/SENG499-S22-Company3/backend/blob/mustafa/dockerfile-modified/package.json](url)
[https://github.com/SENG499-S22-Company3/backend/blob/mustafa/dockerfile-modified/tsconfig.json](url)

aqeelmozumder commented:
I have updated the Dockerfile, modifying code from Mustafa's suggested Dockerfile. Have ran issues with copyfiles dependency and therefore worked with an alternative by copying the schema file to the the container

The Docker-compose file is now updated as well with container and service name postgresql pulling version 14, using admin:admin and default database is now postgresql

aqeelmozumder commented on Dockerfile:
```diff
@@ -0,0 +1,20 @@
+
+RUN npm run build
+
+COPY ./src/schema/schema.graphql /usr/src/app/build/src/schema
```
npm build does not generate the schema.graphql in the build directory, therefore had to copy the schema from src to the build

aomi reviewed (approved):
LGTM
